# Analysis of Ebay Car Sales 

The main aim of this project is to carry out an analysis of car sales from the classsifieds section of eBay Germany.

The data set entails data across these point:
- dateCrawled - When this ad was first crawled. All field-values are taken from this date.
- name - Name of the car.
- seller - Whether the seller is private or a dealer.
- offerType - The type of listing
- price - The price on the ad to sell the car.
- abtest - Whether the listing is included in an A/B test.
- vehicleType - The vehicle Type.
- yearOfRegistration - The year in which the car was first registered.
- gearbox - The transmission type.
- powerPS - The power of the car in PS.
- model - The car model name.
- kilometer - How many kilometers the car has driven.
- monthOfRegistration - The month in which the car was   first registered.
- fuelType - What type of fuel the car uses.
- brand - The brand of the car.
- notRepairedDamage - If the car has a damage which is   not yet repaired.
- dateCreated - The date on which the eBay listing was   created.
- nrOfPictures - The number of pictures in the ad.
- postalCode - The postal code for the location of the   vehicle.
- lastSeenOnline - When the crawler saw this ad last     online.

In [33]:
import pandas as pd
import numpy as np
import re
autos = pd.read_csv('autos.csv', encoding = 'Latin-1')
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

In [6]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

The data set is made of 50000 entries across 20 columns largely populated by strings. Some columns have null values, however non of te columns have more than 20% null values.

The columns names are in camelcase instead of snakecase which would make it a challenge to replace spaces with underscores.

In [10]:
#printing array of columns
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [34]:
autos.columns = ['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'registration_year', 'gearbox', 'powerPS', 'model',
       'odometer', 'registration_month', 'fuelType', 'brand',
       'unrepaired_damage', 'ad_created', 'nrOfPictures', 'postalCode',
       'lastSeen']

# alernative method
def cleaner(col):
    col = col.replace('yearOfRegistration', 'registration_year')
    col = col.replace('monthOfRegistration', 'registration_month')
    col = col.replace('notRepairDamage', 'unrepaired_damage')
    col = col.replace('dateCreated', 'ad_created')
    col = re.sub('(?<!^)(?=[A-Z])', '_', col).lower()
    return col

autos.columns = [cleaner(c) for c in autos.columns]
autos.columns
    

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_p_s', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'nr_of_pictures', 'postal_code',
       'last_seen'],
      dtype='object')

In [16]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  registration_year  \
0  privat   Angebot  $5,000  control         bus               2004   
1  privat   Angebot  $8,500  control   limousine               1997   
2  privat   Angebot  $8,990     test   limousine               2009   
3  privat   Angebot  $4,350  control  kleinwagen               2007   
4  privat   Angebot  $1,350     test       kombi               2003   

     gearbox  powerPS   model   odometer  registration_month fuelType  \
0    manuell      158  andere  150,000km                   3      lpg   
1  automatik      286     7er  150,000km                   6   benzin   
2    manuell      102    golf   70,000km                   7   benzin   
3  automatik       71  fortwo   70,000km                   6   benzin   
4    manuell        0   focus  150,000km                   7   benzin   

        brand unrepaired_damage           ad_created  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

# Initial Exploration & Cleaning of Data

Initial exploration to determine cleaing tasks to be done. Focus is on:
- Text columns where all or almost all values are the same to determine   which to drop since they lack useful information for analysis.
- Numeric data stored as tect which can be cleaned and converted

In [31]:
# Exploring descriptive statistics across all columns
autos.describe(include = 'all')

dateCrawled         name  seller offerType         price  \
count                 50000        50000   50000     50000  5.000000e+04   
unique                48213        38754       2         2           NaN   
top     2016-03-23 19:38:20  Ford_Fiesta  privat   Angebot           NaN   
freq                      3           78   49999     49999           NaN   
mean                    NaN          NaN     NaN       NaN  9.840044e+03   
std                     NaN          NaN     NaN       NaN  4.811044e+05   
min                     NaN          NaN     NaN       NaN  0.000000e+00   
25%                     NaN          NaN     NaN       NaN  1.100000e+03   
50%                     NaN          NaN     NaN       NaN  2.950000e+03   
75%                     NaN          NaN     NaN       NaN  7.200000e+03   
max                     NaN          NaN     NaN       NaN  1.000000e+08   

       abtest vehicleType  registration_year  gearbox       powerPS  model  \
count   50000       44905       50000.000000    47320  50000.000000  47242   
unique      2           8                NaN        2           NaN    245   
top      test   limousine                NaN  manuell           NaN   golf   
freq    25756       12859                NaN    36993           NaN   4024   
mean      NaN         NaN        2005.073280      NaN    116.355920    NaN   
std       NaN         NaN         105.712813      NaN    209.216627    NaN   
min       NaN         NaN        1000.000000      NaN      0.000000    NaN   
25%       NaN         NaN        1999.000000      NaN     70.000000    NaN   
50%       NaN         NaN        2003.000000      NaN    105.000000    NaN   
75%       NaN         NaN        2008.000000      NaN    150.000000    NaN   
max       NaN         NaN        9999.000000      NaN  17700.000000    NaN   

             odometer  registration_month fuelType       brand  \
count    50000.000000        50000.000000    45518       50000   
unique            NaN                 NaN        7          40   
top               NaN                 NaN   benzin  volkswagen   
freq              NaN                 NaN    30107       10687   
mean    125732.700000            5.723360      NaN         NaN   
std      40042.211706            3.711984      NaN         NaN   
min       5000.000000            0.000000      NaN         NaN   
25%     125000.000000            3.000000      NaN         NaN   
50%     150000.000000            6.000000      NaN         NaN   
75%     150000.000000            9.000000      NaN         NaN   
max     150000.000000           12.000000      NaN         NaN   

       unrepaired_damage           ad_created  nrOfPictures    postalCode  \
count              40171                50000       50000.0  50000.000000   
unique                 2                   76           NaN           NaN   
top                 nein  2016-04-03 00:00:00           NaN           NaN   
freq               35232                 1946           NaN           NaN   
mean                 NaN                  NaN           0.0  50813.627300   
std                  NaN                  NaN           0.0  25779.747957   
min                  NaN                  NaN           0.0   1067.000000   
25%                  NaN                  NaN           0.0  30451.000000   
50%                  NaN                  NaN           0.0  49577.000000   
75%                  NaN                  NaN           0.0  71540.000000   
max                  NaN                  NaN           0.0  99998.000000   

                   lastSeen  
count                 50000  
unique                39481  
top     2016-04-07 06:17:27  
freq                      8  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

The column with the number of pictures has no values, lacking useful information and can be dropped.

The minimum and maximum valuesof the regiatration year are 1000 and 9999 respectively, indicating presence of unreliable and flawed data meaning some rows can be safely removed.

The variance betwwen the maximum value  and the value at 75% in the powerPS columns is too high suggesting the presence of inaccurate data in the column.

The minimum registration month is 0, suggesting the presence of inaccurate data in several rows.

The values of the seller and offer type are largely the same so they can be removed without affecting the accuracy of our analysis.

The Price and Odometer columns have numeric values stored as text and these can be converted to integers to enable further analysis


In [20]:
autos['price'].unique()


array(['$5,000', '$8,500', '$8,990', ..., '$385', '$22,200', '$16,995'],
      dtype=object)

In [22]:
autos['odometer'].unique()

array(['150,000km', '70,000km', '50,000km', '80,000km', '10,000km',
       '30,000km', '125,000km', '90,000km', '20,000km', '60,000km',
       '5,000km', '100,000km', '40,000km'], dtype=object)

In [39]:
# Removal of non numeric characters and conversion to numeric dtype

def price_conversion(column):
    if pd.isnull(column) == True:
        return None
    else:
        string = column.split(' ')[0]
        string = string.replace("$", "")
        string = string.replace(",", "")
        return int(string)

autos['price'] = autos['price'].apply(price_conversion)
autos['price'].head()

0    5000
1    8500
2    8990
3    4350
4    1350
Name: price, dtype: int64

In [40]:
def odo_conversion(column):
    if pd.isnull(column) == True:
        return None
    else:
        string = column.split(' ')[0]
        string = string.replace("km", "")
        string = string.replace(",", "")
        return int(string)

autos['odometer'] = autos['odometer'].apply(odo_conversion)
autos['odometer'].head()

0    150000
1    150000
2     70000
3     70000
4    150000
Name: odometer, dtype: int64

In [61]:
#renaming of odometer column to indicate figures are in km
autos = autos.rename(columns = {'odometer':'odometer_km'})

In [62]:
# removal of irrelevant rows
autos = autos.drop(['nr_of_pictures', 'seller', "offer_type"], axis=1)

Contitnued exploraion of the data to identify data that doesn't fit. 
Analysis of price and odometer_km columns:
- looking for any outlying values that are unrealistically high or low that should be removed.

In [66]:
autos['odometer_km'].unique().shape


(13,)

In [67]:
autos['odometer_km'].describe()


count     48565.000000
mean     125770.101925
std       39788.636804
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [69]:
autos['odometer_km'].value_counts()

150000    31414
125000     5057
100000     2115
90000      1734
80000      1415
70000      1217
60000      1155
50000      1012
5000        836
40000       815
30000       780
20000       762
10000       253
Name: odometer_km, dtype: int64

The odometer_km column has 13 unique values. The reading do not seem to be unrealistically high or low.

In [60]:
autos['price'].unique().shape


(2346,)

In [46]:
autos['price'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [49]:
autos['price'].value_counts().head(20)

0       1421
500      781
1500     734
2500     643
1000     639
1200     639
600      531
800      498
3500     498
2000     460
999      434
750      433
900      420
650      419
850      410
700      395
4500     394
300      384
2200     382
950      379
Name: price, dtype: int64

There are 1421 rows of cars with a price of \$0. These rows can be removed without compromising the quality of the analysis. The maximum price is \$100,000,000 which needs to be explored further since it is extreemely high and seems unlikely.

In [52]:
autos['price'].value_counts().sort_index(ascending=False).head(15)

99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
350000      1
345000      1
299000      1
295000      1
265000      1
Name: price, dtype: int64

In [53]:
autos['price'].value_counts().sort_index(ascending=True).head(15)

0     1421
1      156
2        3
3        1
5        2
8        1
9        1
10       7
11       2
12       3
13       2
14       1
15       2
17       3
18       1
Name: price, dtype: int64

The jump in prices from \$350k \$999k isquite significant and the variance betwwen prices as they go higher becomes uite significant. To maintain the integrity of the analyses, it would be safe to remove rows with prices above \$350k.

There are a number of cars priced at \$18 and below. Despite being uncharacteristically low, the pricing at the moment of obtaingin the data would make sense since ebay is an auction site and bidding prices could start at \$1.

In [55]:
# rows of data to keep
autos = autos[autos['price'].between(1,350000)]
autos['price'].describe()

count     48565.000000
mean       5888.935591
std        9059.854754
min           1.000000
25%        1200.000000
50%        3000.000000
75%        7490.000000
max      350000.000000
Name: price, dtype: float64

There are 5 columns that should represent date values

- `date_crawled`: added by the crawler
- `last_seen`: added by the crawler
- `ad_created`: from the website
- `registration_month`: from the website
- `registration_year`: from the website

date_crawled, last_seen and ad_created columns are all popuulated by strings and need to be converted to numerical representation so as to understand them quantitatively.

In [75]:
# formatting of columns
autos[['date_crawled', 'ad_created', 'last_seen']][0:5]

date_crawled           ad_created            last_seen
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4  2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50

Calculating Distribution of Values as Percentages while ranking them in acsending order

In [72]:
autos['date_crawled'].str[:10].value_counts(normalize = True, dropna=False).sort_index()

2016-03-05    0.025327
2016-03-06    0.014043
2016-03-07    0.036014
2016-03-08    0.033296
2016-03-09    0.033090
2016-03-10    0.032184
2016-03-11    0.032575
2016-03-12    0.036920
2016-03-13    0.015670
2016-03-14    0.036549
2016-03-15    0.034284
2016-03-16    0.029610
2016-03-17    0.031628
2016-03-18    0.012911
2016-03-19    0.034778
2016-03-20    0.037887
2016-03-21    0.037373
2016-03-22    0.032987
2016-03-23    0.032225
2016-03-24    0.029342
2016-03-25    0.031607
2016-03-26    0.032204
2016-03-27    0.031092
2016-03-28    0.034860
2016-03-29    0.034099
2016-03-30    0.033687
2016-03-31    0.031834
2016-04-01    0.033687
2016-04-02    0.035478
2016-04-03    0.038608
2016-04-04    0.036487
2016-04-05    0.013096
2016-04-06    0.003171
2016-04-07    0.001400
Name: date_crawled, dtype: float64

The distribution of listings crawled is uniform across the various dates

In [73]:
autos['ad_created'].str[:10].value_counts(normalize = True, dropna=False).sort_index()

2015-06-11    0.000021
2015-08-10    0.000021
2015-09-09    0.000021
2015-11-10    0.000021
2015-12-05    0.000021
2015-12-30    0.000021
2016-01-03    0.000021
2016-01-07    0.000021
2016-01-10    0.000041
2016-01-13    0.000021
2016-01-14    0.000021
2016-01-16    0.000021
2016-01-22    0.000021
2016-01-27    0.000062
2016-01-29    0.000021
2016-02-01    0.000021
2016-02-02    0.000041
2016-02-05    0.000041
2016-02-07    0.000021
2016-02-08    0.000021
2016-02-09    0.000021
2016-02-11    0.000021
2016-02-12    0.000041
2016-02-14    0.000041
2016-02-16    0.000021
2016-02-17    0.000021
2016-02-18    0.000041
2016-02-19    0.000062
2016-02-20    0.000041
2016-02-21    0.000062
                ...   
2016-03-09    0.033151
2016-03-10    0.031895
2016-03-11    0.032904
2016-03-12    0.036755
2016-03-13    0.017008
2016-03-14    0.035190
2016-03-15    0.034016
2016-03-16    0.030125
2016-03-17    0.031278
2016-03-18    0.013590
2016-03-19    0.033687
2016-03-20    0.037949
2016-03-21 

The ads were created during the same time period. Beyond a certain date there is a low frequency of listings or ads created probably because the auctions are over and completed. 

In [74]:
autos['last_seen'].str[:10].value_counts(normalize = True, dropna=False).sort_index()

2016-03-05    0.001071
2016-03-06    0.004324
2016-03-07    0.005395
2016-03-08    0.007413
2016-03-09    0.009595
2016-03-10    0.010666
2016-03-11    0.012375
2016-03-12    0.023783
2016-03-13    0.008895
2016-03-14    0.012602
2016-03-15    0.015876
2016-03-16    0.016452
2016-03-17    0.028086
2016-03-18    0.007351
2016-03-19    0.015834
2016-03-20    0.020653
2016-03-21    0.020632
2016-03-22    0.021373
2016-03-23    0.018532
2016-03-24    0.019767
2016-03-25    0.019211
2016-03-26    0.016802
2016-03-27    0.015649
2016-03-28    0.020859
2016-03-29    0.022341
2016-03-30    0.024771
2016-03-31    0.023783
2016-04-01    0.022794
2016-04-02    0.024915
2016-04-03    0.025203
2016-04-04    0.024483
2016-04-05    0.124761
2016-04-06    0.221806
2016-04-07    0.131947
Name: last_seen, dtype: float64

With the last seen being the last time the crawler saw the ad online, the spike during the last three days shows that there was a relatively higher number of listings expired during the last three days.

The days leading up to the last three days had a normal distribution with no increase expiring listings so a surge in sales during the last three days is doubtful but still possible.

Exploring Registration Years

There were outlying values in terms of the regostration years of vehicles with some being far into the past before vehicles were invented and the rest being into the future beyond the present time period.

The next step would be to remove rows with vehicle registration dates earlier then when the cars were made and those beyond the latest dates of the dataset.

Cars available for public use en masse existed largely during the onset of the 20th century so all registration years dates prior to 1900 are invalid as are those with registration dates beyond the latest date of the date set 2016.

In [76]:
autos['registration_year'].describe()

count    48565.000000
mean      2004.755421
std         88.643887
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

In [82]:
# fraction of cars in dataset registered between 1900 and 2016
accurate_dates = (autos['registration_year'].between(1900,2016).sum()/autos.shape[0]) * 100
print(accurate_dates)

96.1206630289303


The cars with valid dates account for 96% of the cars in the dataset meaning the non valid rows, are at most 4% of the dataset.

In [87]:
# removal of dataset outisde specified dates and calculating distribution
autos = autos[autos['registration_year'].between(1900,2016)]
autos['registration_year'].value_counts(normalize=True).sort_index()

1910    0.000107
1927    0.000021
1929    0.000021
1931    0.000021
1934    0.000043
1937    0.000086
1938    0.000021
1939    0.000021
1941    0.000043
1943    0.000021
1948    0.000021
1950    0.000064
1951    0.000043
1952    0.000021
1953    0.000021
1954    0.000043
1955    0.000043
1956    0.000086
1957    0.000043
1958    0.000086
1959    0.000129
1960    0.000493
1961    0.000129
1962    0.000086
1963    0.000171
1964    0.000257
1965    0.000364
1966    0.000471
1967    0.000557
1968    0.000557
          ...   
1987    0.001542
1988    0.002892
1989    0.003727
1990    0.007433
1991    0.007262
1992    0.007926
1993    0.009104
1994    0.013474
1995    0.026285
1996    0.029412
1997    0.041794
1998    0.050620
1999    0.062060
2000    0.067608
2001    0.056468
2002    0.053255
2003    0.057818
2004    0.057904
2005    0.062895
2006    0.057197
2007    0.048778
2008    0.047450
2009    0.044665
2010    0.034040
2011    0.034768
2012    0.028063
2013    0.017202
2014    0.0142

From the distribution of the vehicles across the years of registration, fewer cars listed for auction were made in the earlier years of the 20th century with the most early being 1910. The proportion of cars registered gradually increases among the years later into the 20th century and peak during the early years of the 21s century with the number of listed cars registered in the 21st century declining - year on year - beyond the year 2005.

Most of the cars listed were registered between the years 2000 - 2007

EXPLORATION OF THE VEHICLE BRANDS

In [89]:
autos['brand'].value_counts(normalize = True)

volkswagen        0.211264
bmw               0.110045
opel              0.107581
mercedes_benz     0.096463
audi              0.086566
ford              0.069900
renault           0.047150
peugeot           0.029841
fiat              0.025642
seat              0.018273
skoda             0.016409
nissan            0.015274
mazda             0.015188
smart             0.014160
citroen           0.014010
toyota            0.012703
hyundai           0.010025
sonstige_autos    0.009811
volvo             0.009147
mini              0.008762
mitsubishi        0.008226
honda             0.007840
kia               0.007069
alfa_romeo        0.006641
porsche           0.006127
suzuki            0.005934
chevrolet         0.005698
chrysler          0.003513
dacia             0.002635
daihatsu          0.002506
jeep              0.002271
subaru            0.002142
land_rover        0.002099
saab              0.001649
jaguar            0.001564
daewoo            0.001500
trabant           0.001392
r

Most cars listed are of the Volkswagen brand, follwed by Bmw, Opel, Mercedes Benz & Audi, ccollectively making the top five most represented brands among the listings. These brands are also all German manufactured brands.

In [91]:
brands = autos['brand'].value_counts(normalize = True)

most_common_brands = brands[brands > 0.05].index
most_common_brands

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford'], dtype='object')

The brands hat individually make up more than 5% of the listings are Volkswagen, Bmw, Opel, Mercedes Benz, Audi and Ford.

In [96]:
brands_avg_price = {}

for brands in most_common_brands:
    brandd = autos[autos['brand'] == brands]
    avg_price = brandd['price'].mean()
    brands_avg_price[brands] = int(avg_price)
    
brands_avg_price    

{'audi': 9336,
 'bmw': 8332,
 'ford': 3749,
 'mercedes_benz': 8628,
 'opel': 2975,
 'volkswagen': 5402}

The top brands have varying prices with Audi, Mercedes Benza and Bmw being the most expensive, Volkswagen having mid-priced vehicles and Opel and Ford being the cheapest vehicles.

In [105]:
brand_mean_milage = {}

for brands in most_common_brands:
    brandd = autos[autos['brand'] == brands]
    avg_milage = brandd['odometer_km'].mean()
    brand_mean_milage[brands] = int(avg_milage)

brand_mean_milage

{'audi': 129157,
 'bmw': 132572,
 'ford': 124266,
 'mercedes_benz': 130788,
 'opel': 129310,
 'volkswagen': 128707}

In [114]:
mean_milage = pd.Series(brand_mean_milage).sort_values(ascending = False)
mean_price = pd.Series(brands_avg_price).sort_values(ascending = False)

In [115]:
top_brand_info = pd.DataFrame(mean_milage, columns = ['mean_milage'])
top_brand_info

mean_milage
bmw                 132572
mercedes_benz       130788
opel                129310
audi                129157
volkswagen          128707
ford                124266

In [116]:
top_brand_info['mean_price'] = mean_price
top_brand_info

mean_milage  mean_price
bmw                 132572        8332
mercedes_benz       130788        8628
opel                129310        2975
audi                129157        9336
volkswagen          128707        5402
ford                124266        3749

The average milage does not really vary much across the top five brands ranging between 124000 - 13300. Bmw cars have the highest milage on average while ford casrs have the least milage on average. The expensive cars have a high average milage with the lower priced ones having lower average milages with the exclusion of Opel which has a relatively higher margin.

In [117]:
autos['gearbox'].value_counts()

manuell      34715
automatik     9856
Name: gearbox, dtype: int64

In [120]:
autos['unrepaired_damage'].value_counts()

nein    33834
ja       4540
Name: unrepaired_damage, dtype: int64

In [122]:
autos['fuel_type'].value_counts()

benzin     28540
diesel     14032
lpg          649
cng           71
hybrid        37
elektro       19
andere        15
Name: fuel_type, dtype: int64

In [125]:
autos['vehicle_type'].value_counts()

limousine     12598
kleinwagen    10585
kombi          8930
bus            4031
cabrio         3016
coupe          2462
suv            1965
andere          390
Name: vehicle_type, dtype: int64

In [130]:
# mapping german values to english
gearbox = {'manuell': 'manual', 'automatik':'automatic'}
unrepaired_damage = {'nein':'no','ja':'yes'}

autos['gearbox'] = autos['gearbox'].map(gearbox)
autos['unrepaired_damage'] = autos['unrepaired_damage'].map(unrepaired_damage)
autos['fuel_type'] = autos['fuel_type'].replace(['benzin','elektro','andere'], ['gasoline','electric','other'])
autos['vehicle_type'] = autos['vehicle_type'].replace(['limousine','kleinwagen','kombi','cabrio','andere'],['sedan','small car','stationwagen','convertible','other'])
autos['model'] = autos['model'].replace('andere','other')
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   price   abtest  vehicle_type  registration_year gearbox  power_p_s   model  \
0   5000  control           bus               2004     NaN        158   other   
1   8500  control         sedan               1997     NaN        286     7er   
2   8990     test         sedan               2009     NaN        102    golf   
3   4350  control     small car               2007     NaN         71  fortwo   
4   1350     test  stationwagen               2003     NaN          0   focus   

   odometer_km  registration_month fuel_type       brand unrepaired_damage  \
0       150000                   3       lpg     peugeot               NaN   
1       150000                   6  gasoline         bmw               NaN   
2        70000                   7  gasoline  volkswagen               NaN   
3        70000                   6  gasoline       smart               NaN   
4       150000                   7  gasoline        ford               NaN   

            ad_created  postal_code            last_seen  
0  2016-03-26 00:00:00        79588  2016-04-06 06:45:54  
1  2016-04-04 00:00:00        71034  2016-04-06 14:45:08  
2  2016-03-26 00:00:00        35394  2016-04-06 20:15:37  
3  2016-03-12 00:00:00        33729  2016-03-15 03:16:28  
4  2016-04-01 00:00:00        39218  2016-04-01 14:38:50